### PR TITLE
[SW-2084] Deprecate DeepLearningSupport and GBMSupport in favor of SW Algo API

### PR DESCRIPTION
**Pull request by jakubhava**

### CODE REVIEW

File: doc/src/site/sphinx/migration_guide.rst [hunk 1]
```diff
@@ -21,6 +21,10 @@ From 3.30 to 3.32
 - The ``setH2OCluster(ip, port)`` method on ``H2OConf`` in all APIs doesn't implicitly set the external backend anymore.
   The method ``setExternalClusterMode()`` must be called explicitly.
 
+- The method ``DLModel`` in ``water.support.DeepLearningSupport`` is removed. Please use Sparkling Water algorithm API
```
Comment (mn-mikke) [inline doc/src/site/sphinx/migration_guide.rst]: I would maybe mention that it's related only to Scala API.

File: core/src/main/scala/water/support/DeepLearningSupport.scala [hunk 2]
```diff
@@ -42,6 +44,7 @@ trait DeepLearningSupport {
     * @tparam T H2O Frame Type
     * @return Deep Learning Model
     */
+  @DeprecatedMethod("Sparkling Water Algorithm API to train and score H2O models", "3.32")
```
Comment (mn-mikke) [inline core/src/main/scala/water/support/DeepLearningSupport.scala]: Maybe, I would add reference to the `H2ODeepLearning` class.

File: core/src/main/scala/water/support/GBMSupport.scala [hunk 3]
```diff
@@ -39,6 +41,7 @@ trait GBMSupport {
     * @tparam T H2O Frame Type
     * @return Gradient Boosting Model
     */
+  @DeprecatedMethod("Sparkling Water Algorithm API to train and score H2O models", "3.32")
```
Comment (mn-mikke) [inline core/src/main/scala/water/support/GBMSupport.scala]: Maybe, I would add reference to the `H2OGBM` class.

File: doc/src/site/sphinx/migration_guide.rst (hunk 1 above)
Comment (jakubhava) [reply doc/src/site/sphinx/migration_guide.rst]: 👍

File: core/src/main/scala/water/support/DeepLearningSupport.scala (hunk 2 above)
Comment (jakubhava) [reply core/src/main/scala/water/support/DeepLearningSupport.scala]: 👍

File: core/src/main/scala/water/support/GBMSupport.scala (hunk 3 above)
Comment (jakubhava) [reply core/src/main/scala/water/support/GBMSupport.scala]: 👍

**Pull request merged**